File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    col = {"code": 0, "date": 1, "maximum": 2, "remain": 3}
    
    filtered_data = [item for item in data if item[col[ext]] < val_ext]
    
    filtered_data.sort(key=lambda x: x[col[sort_by]])
    
    return filtered_data

File: 프로그래머스/lv1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    unordered_map<string, int> col;
    col["code"] = 0;
    col["date"] = 1;
    col["maximum"] = 2;
    col["remain"] = 3;
    
    vector<vector<int>> filter;
    for (const auto& d : data) {
        if (d[col[ext]] < val_ext) {
            filter.push_back(d);
        }
    }
    
    sort(filter.begin(), filter.end(), [&col, sort_by](const vector<int>& a, const vector<int>& b) {
        return a[col[sort_by]] < b[col[sort_by]];
    });
    
    return filter;
}